DXRFixes/DeusEx/Classes/Carcass.uc

```diff
@@ -316,10 +312,43 @@ function bool TryLootItem(DeusExPlayer player, Inventory item)
         }
     }
 
-    if (item.IsA('DeusExAmmo'))
+    if (item.IsA('Ammo'))
     {
-        if (DeusExAmmo(item).AmmoAmount == 0)
+        if (!dropsAmmo || Ammo(item).AmmoAmount == 0) {
+            DeleteInventory(item);
+            item.Destroy();
             return false;
+        }
+
+        playerAmmo = Ammo(player.FindInventoryType(item.class));
+
+        ammoAdded = Min( Ammo(item).AmmoAmount, Ammo(item).MaxAmmo - playerAmmo.AmmoAmount );
```

Comment:
missing check for playerAmmo != None

also you might want to do `AmmoType = Ammo(item);` to reduce how often you have to cast that variable

Reply:
I don't check for None because playerAmmo.AmmoAmount is zero if it doesn't exist. I'll grant that it's questionable.

The "AmmoType" name bugs me because it's an object, not a type. `AmmoType.AmmoAmount == 0` doesn't really make sense when you read it. In fact, it made deciphering the code above it move difficult for me. I considered creating a new variable for it but decided that replacing five typecasts with one wasn't worth it, and sticking with "item" makes what's going on more clear.

Reply:
I think a new variable name would be fine

also the missing check if playerAmmo exists or not would cause a warning in the log file here because playerAmmo.AmmoAmount would be an "Accessed None" error